src/settings/global_user.rs

```diff
@@ -20,16 +21,33 @@ use std::io::Write;
 #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 #[serde(untagged)]
 pub enum GlobalUser {
-    TokenAuth { api_token: String },
-    GlobalKeyAuth { email: String, api_key: String },
+    ApiTokenAuth {
+        api_token: String,
+    },
+    OAuthTokenAuth {
+        oauth_token: String,
+        refresh_token: String,
+        expiration_time: String,
+    },
+    GlobalKeyAuth {
+        email: String,
+        api_key: String,
+    },
 }
 
 impl GlobalUser {
     pub fn new() -> Result<Self> {
         let environment = Environment::with_whitelist(ENV_VAR_WHITELIST.to_vec());
 
         let config_path = get_global_config_path();
-        GlobalUser::build(environment, config_path)
+        let mut new_user = GlobalUser::build(environment, config_path);
+
+        // Check if oauth token is expired
+        if let Ok(ref mut oauth_user) = new_user {
+            let _res =
+                check_update_oauth_token(oauth_user).expect("Failed to refresh access token");
```

Comment:
Another case to consider more useful error handling / context-sharing with the end user.